src/types.ts

```diff
@@ -152,24 +157,31 @@ export interface BinaryRecordedEvent extends RecordedEventBase {
   metadata: Uint8Array;
 }
 
-export interface AllStreamJSONRecordedEvent extends JSONRecordedEvent {
+export interface AllStreamJSONRecordedEvent<
+  EventType extends string = string,
+  Data = unknown,
+  Metadata extends Record<string, unknown> = Record<string, unknown>
+> extends JSONRecordedEvent<EventType, Data, Metadata> {
   /**
    * Position of this event in the transaction log.
    */
   position: Position;
 }
 
-export interface AllStreamBinaryRecordedEvent extends BinaryRecordedEvent {
+export interface AllStreamBinaryRecordedEvent<EventType extends string = string>
+  extends BinaryRecordedEvent<EventType> {
   /**
    * Position of this event in the transaction log.
    */
   position: Position;
 }
 
-export type RecordedEvent = JSONRecordedEvent | BinaryRecordedEvent;
+export type RecordedEvent =
+  | JSONRecordedEvent<string>
+  | BinaryRecordedEvent<string>;
 export type AllStreamRecordedEvent =
-  | AllStreamJSONRecordedEvent
-  | AllStreamBinaryRecordedEvent;
+  | AllStreamJSONRecordedEvent<string>
+  | AllStreamBinaryRecordedEvent<string>;
```

Comment:
`string` is the default no?

Reply:
Ah yes - fixed in e81a65d6e7bf273ebf9852e148f85812a14112cc

Reply:
You missed `AllStreamRecordedEvent`

Reply:
Aargh. Of course I did. 5336257c7eae478509223f7b5d7e745434360144